DirectProgramming/C++SYCL_FPGA/Tutorials/Tools/experimental/platform_designer/add-oneapi/CMakeLists.txt

```diff
@@ -82,6 +82,8 @@ message(STATUS "Additional USER_LIB_PATHS=${USER_LIB_PATHS}")
 link_libraries(${USER_LIBS})
 message(STATUS "Additional USER_LIBS=${USER_LIBS}")
 
+include_directories("/opt/intel/oneapi/compiler/2023.0.0/linux/lib/oclfpga/include")
```

Comment:
There is a better way to solve this issue, which is not going to be release dependent:
```
# This is for finding the install path to the necessary include files
if(DEFINED ENV{INTELFPGAOCLSDKROOT})
   set(SDK_ROOT_PATH $ENV{INTELFPGAOCLSDKROOT})
else()
   message(FATAL_ERROR "The INTELFPGAOCLSDKROOT environment variable must be defined for this tutorial to compile.")
endif()
```
and then add `${SDK_ROOT_PATH}/include` and `${SDK_ROOT_PATH}/include/sycl/ext/intel/prototype` to the `USER_INCLUDE_PATHS` variable.

I'll make some tests on my end

Reply:
Makes sense!

Reply:
I've been running some tests in a 2023.0 release, and I don't need this include.

```
C:\Users\yuguen\Downloads\oneAPI-samples\DirectProgramming\C++SYCL_FPGA\Tutorials\Tools\experimental\platform_designer\add-oneapi\b>cmake .. -G "NMake Makefiles"
CMake Deprecation Warning at C:/Program Files/Microsoft Visual Studio/2022/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.21/Modules/CMakeForceCompiler.cmake:89 (message):
  The CMAKE_FORCE_CXX_COMPILER macro is deprecated.  Instead just set
  CMAKE_CXX_COMPILER and allow CMake to identify the compiler.
Call Stack (most recent call first):
  CMakeLists.txt:7 (CMAKE_FORCE_CXX_COMPILER)


-- Configuring the design to run on FPGA board 10AS066N3F40E2SG
-- Additional USER_FPGA_FLAGS=
-- Additional USER_FLAGS=
-- Additional USER_INCLUDE_PATHS=../../../../../include
-- Additional USER_LIB_PATHS=
-- Additional USER_LIBS=
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/yuguen/Downloads/oneAPI-samples/DirectProgramming/C++SYCL_FPGA/Tutorials/Tools/experimental/platform_designer/add-oneapi/b

C:\Users\yuguen\Downloads\oneAPI-samples\DirectProgramming\C++SYCL_FPGA\Tutorials\Tools\experimental\platform_designer\add-oneapi\b>nmake fpga_emu

Microsoft (R) Program Maintenance Utility Version 14.30.30706.0
Copyright (C) Microsoft Corporation.  All rights reserved.

[ 33%] To compile manually:
C:/Program Files (x86)/Intel/oneAPI/compiler/latest/windows/bin/icx-cl.exe -I../../../../../../include -fsycl -fintelfpga -Wall /EHsc -Qactypes -DFPGA_EMULATOR -c ../src/add.cpp -o CMakeFiles/fpga_emu.dir/src/add.cpp.obj

To link manually:
C:/Program Files (x86)/Intel/oneAPI/compiler/latest/windows/bin/icx-cl.exe -fsycl -fintelfpga -Qactypes -o add.fpga_emu.exe CMakeFiles/fpga_emu.dir/src/add.cpp.obj
[ 33%] Built target displayEmulationCompileCommands
Scanning dependencies of target fpga_emu
[ 66%] Building CXX object CMakeFiles/fpga_emu.dir/src/add.cpp.obj
Intel(R) oneAPI DPC++/C++ Compiler for applications running on Intel(R) 64, Version 2023.0.0 Build 20221201
Copyright (C) 1985-2022 Intel Corporation. All rights reserved.

[100%] Linking CXX executable add.fpga_emu.exe
Intel(R) oneAPI DPC++/C++ Compiler for applications running on Intel(R) 64, Version 2023.0.0 Build 20221201
Copyright (C) 1985-2022 Intel Corporation. All rights reserved.

[100%] Built target fpga_emu
```

This is using the oneapi DPC++/C++ compiler on Windows, downloaded on the 16th of March 2023.
So I'll wait for you to confirm that you had correctly called `setvars.sh` before running the compile.
Thanks!

Reply:
> This is using the oneapi DPC++/C++ compiler on Windows, downloaded on the 16th of March 2023.
So I'll wait for you to confirm that you had correctly called setvars.sh before running the compile.
Thanks!

@yuguen-intel, using Linux (Ubuntu 22.04) and `2023.0`. `setvars.sh` is properly sourced. Still failing for me.

In  my setup, I need to add this up or it fails. I don't have the bandwidth this week to build a reproducible setup but if you're fine waiting, I can try and block a couple of hours next week to build a Docker container that allows you to reproduce it easily.

Reply:
No worries, I'll get myself a Linux 2023.0 and test it as well, many thanks!

Reply:
On a fresh 2023.0 Linux install (Ubuntu 22.04):
```
yohann@yuguen-mobl2:/mnt/c/Users/yuguen/Downloads/oneAPI-samples/DirectProgramming/C++SYCL_FPGA/Tutorials/Tools/experimental/platform_designer/add-oneapi/c$ cmake ..
-- The CXX compiler identification is Clang 16.0.0
-- Check for working CXX compiler: /opt/intel/oneapi/compiler/2023.0.0/linux/bin/icpx
-- Check for working CXX compiler: /opt/intel/oneapi/compiler/2023.0.0/linux/bin/icpx -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring the design to run on FPGA board 10AS066N3F40E2SG
-- Additional USER_FPGA_FLAGS=
-- Additional USER_FLAGS=
-- Additional USER_INCLUDE_PATHS=../../../../../include
-- Additional USER_LIB_PATHS=
-- Additional USER_LIBS=
-- Configuring done
-- Generating done
-- Build files have been written to: /mnt/c/Users/yuguen/Downloads/oneAPI-samples/DirectProgramming/C++SYCL_FPGA/Tutorials/Tools/experimental/platform_designer/add-oneapi/c
yohann@yuguen-mobl2:/mnt/c/Users/yuguen/Downloads/oneAPI-samples/DirectProgramming/C++SYCL_FPGA/Tutorials/Tools/experimental/platform_designer/add-oneapi/c$ make fpga_emu
Scanning dependencies of target displayEmulationCompileCommands
[ 33%] To compile manually:
/opt/intel/oneapi/compiler/2023.0.0/linux/bin/icpx -I../../../../../../include -fsycl -fintelfpga -Wall -qactypes -DFPGA_EMULATOR -c ../src/add.cpp -o CMakeFiles/fpga_emu.dir/src/add.cpp.o

To link manually:
/opt/intel/oneapi/compiler/2023.0.0/linux/bin/icpx -fsycl -fintelfpga -qactypes -o add.fpga_emu CMakeFiles/fpga_emu.dir/src/add.cpp.o
[ 33%] Built target displayEmulationCompileCommands
Scanning dependencies of target fpga_emu
[ 66%] Building CXX object CMakeFiles/fpga_emu.dir/src/add.cpp.o
[100%] Linking CXX executable add.fpga_emu
[100%] Built target fpga_emu
```